src/opentimelineio/typeRegistry.cpp

```diff
@@ -113,6 +111,33 @@ TypeRegistry::TypeRegistry()
 
         d->erase("media_reference");
     });
+
+    // 2->1
+    register_downgrade_function(Clip::Schema::name, 2, [](AnyDictionary* d) {
+        AnyDictionary media_refs = d->get_default("media_references", AnyDictionary());
+        const std::string active_reference_key = (
+                d->get_default("active_media_reference_key", std::string(""))
+        );
+        AnyDictionary active_ref = AnyDictionary();
+        if (active_reference_key != "") {
+            active_ref = media_refs.get_default(active_reference_key, AnyDictionary());
+        }
+
+        (*d)["media_reference"] = active_ref;
+
+        auto downgrade_md = AnyDictionary();
+        downgrade_md["media_references"] = media_refs;
+        downgrade_md["active_media_reference_key"] = active_reference_key;
+
+        d->set_default(
```

Comment:
If information that is discarded during downgrading is going to preserved, I wonder if it would be worth defining a schema for that purpose so there's a standardized way to doing it.

```
"metadata": {}.
"downgrades" : [ 
   {
    "OTIO_SCHEMA": "Downgrade.1",
    "from": "Clip.2",
    "to": "Clip.1",
    "discarded": {
      "media_refererences": media_references
      "active_media_reference": active_media_reference,
      "enabled": true
    }
  }
]
```
If we ever wanted to do something with the discarded data in the future (reconstructing the upgraded version), something well defined would make it easier to do so imo rather than relying on the metadata which would be mixed in with any client metadata.

Just some food for thought.

Reply:
I think initially I want to start with some simpler breadcrumbs (a log in the metadata saying "this was downgraded from version X on day Y").  I'm open to trying this approach but I suspect it might be problematic and not straightforward to reconstruct versions (if there is a conflict between something new authored and old-out of date data that has been kept along, etc.).

Reply:
We talked about this internally and are proposing that this becomes a follow up issue.  This would add further performance overhead and could probably use discussion, so I'll file an issue for it and leave it to a later date.